vochain/state/eventlistener.go

```diff
@@ -28,7 +28,7 @@ type EventListener interface {
 	OnProcessResults(pid []byte, results *models.ProcessResult, txIndex int32)
 	OnProcessesStart(pids [][]byte)
 	OnSetAccount(addr []byte, account *Account)
-	OnTransferTokens(from, to []byte, amount uint64)
+	OnTransferTokens(tx *vochaintx.TransferTokensMeta)
```

Comment:
Why `meta`? Looks like the main type is defined in the indexer in this is a subset. That does not look very correct according to the actual hierarchy.

Here an idea: Consider calling the vochaintx type just "TokenTransfer". Then the indexer type could be named something like "TokenTranferMeta" and would heritage from this type (and add the extra fields required by the indexer).

```
type TokenTransferMeta struct {
   vochaintx.TokenTransfer
   extraFields...
}
 ```

Reply:
I've renamed the types, but I did not use inheritance given that the inner types are different for convenience (i.e Address as common.Address vs Address as types.HexBytes)